internal/scylla/client.go

```diff
@@ -136,10 +136,43 @@ func (c *Client) doHttpRequest(req *http.Request) (resp *http.Response, temporar
 		return
 	}
 
-	temporaryErr = resp.StatusCode == http.StatusBadGateway || resp.StatusCode == http.StatusGatewayTimeout || resp.StatusCode == http.StatusTooManyRequests
+	if getErrorCode(resp) == 000001 {
+		return resp, true, fmt.Errorf("too many request response")
+	}
+	temporaryErr = temporaryErr || resp.StatusCode == http.StatusBadGateway || resp.StatusCode == http.StatusGatewayTimeout || resp.StatusCode == http.StatusTooManyRequests
 	return
 }
 
+func getBodyData(resp *http.Response) []byte {
+	if resp.Body == nil {
+		return nil
+	}
+	data, _ := io.ReadAll(resp.Body)
+	resp.Body.Close()
+	resp.Body = io.NopCloser(bytes.NewReader(data))
+	return data
+}
+
+func getErrorCode(resp *http.Response) int {
+	body := getBodyData(resp)
+	if len(body) == 0 {
+		return 0
+	}
+
+	var data = struct {
+		Error string `json:"error"`
+	}{}
+
+	if err := json.Unmarshal(body, &data); err != nil {
+		return 0
```

Comment:
please don't ignore errors, it could introduce hard to track bugs

please always propagate the error up

Reply:
Not every response carry json, it is going to fail in legit responses, say 404.